drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioSimulationXMLPersistence.java

```diff
@@ -35,6 +36,8 @@ public static ScenarioSimulationXMLPersistence getInstance() {
     }
 
     public String marshal(ScenarioSimulationModel sc) {
+        // force factMappingValues sorting based on factMappings before save
+        sc.getSimulation().getScenarios().forEach(Scenario::sort);
```

Comment:
I would test this and move the comment to the test. Most likely who ever breaks this ignores the comment, but can't ignore a failing test.

Reply:
Removed because not really needed for now